services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProviderFactory.java

```diff
@@ -152,6 +158,18 @@ public SAMLIdentityProviderConfig createConfig() {
                             samlIdentityProviderConfig.setEncryptionPublicKey(defaultCertificate);
                         }
                     }
+                    
+                    samlIdentityProviderConfig.setEnabledConfig(entityType.getValidUntil() == null || entityType.getValidUntil().toGregorianCalendar().getTime().after(new Date()) );
```

Comment:
Time needs to be adjusted to local "Keycloak server" time.
```suggestion
                    samlIdentityProviderConfig.setEnabledConfig(entityType.getValidUntil() == null || entityType.getValidUntil().toGregorianCalendar().getTime().after(new Date(Time.currentTime())) );
```

Reply:
new Date() return Keycloak server local Date time.
Time class you mean org.keycloak.common.util.Time? This method return second and Java Date(long date) wait for milliseconds.
If you insist I could write it as :
 samlIdentityProviderConfig.setEnabledConfig(entityType.getValidUntil() == null || entityType.getValidUntil().toGregorianCalendar().getTime().after(new Date(Time.currentTime() * 1000)) );

Reply:
Yes please, use `currentTimeMillis()` instead.